tests/push/test_push_rule_evaluator.py

```diff
@@ -102,19 +104,19 @@ def test_user_mentions(self) -> None:
         condition = {"kind": "org.matrix.msc3952.is_user_mention"}
 
         # No mentions shouldn't match.
-        evaluator = self._get_evaluator({})
+        evaluator = self._get_evaluator({}, has_mentions=True)
         self.assertFalse(evaluator.matches(condition, "@user:test", None))
 
         # An empty set shouldn't match
-        evaluator = self._get_evaluator({}, user_mentions=set())
+        evaluator = self._get_evaluator({}, has_mentions=True, user_mentions=set())
         self.assertFalse(evaluator.matches(condition, "@user:test", None))
 
         # The Matrix ID appearing anywhere in the mentions list should match
-        evaluator = self._get_evaluator({}, user_mentions={"@user:test"})
+        evaluator = self._get_evaluator({}, has_mentions=True, user_mentions={"@user:test"})
         self.assertTrue(evaluator.matches(condition, "@user:test", None))
 
         evaluator = self._get_evaluator(
-            {}, user_mentions={"@another:test", "@user:test"}
+            {}, has_mentions=True, user_mentions={"@another:test", "@user:test"}
         )
         self.assertTrue(evaluator.matches(condition, "@user:test", None))
```

Comment:
Should we check that the evaluator doesn't match the condition if `has_mentions is False`? Same for `test_room_mentions` below.

Reply:
Yes, but we need to do it in the `BulkPushRuleEvaluator` since that tests the base rules, this tests just specific custom rules.